File: 프로그래머스/lv2/김미래/카펫.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int length = (brown - 4) /2 ; // 노란색의 가로 + 세로 길이
    int garo = 0;
    
    for(int i=1; i<length; i++){
        garo = length - i;
        if(garo * i == yellow){
            break;
        }
    }
    
    if(garo > length-garo){
        answer.push_back(garo+2);
        answer.push_back(length-garo+2);
    }else{
        answer.push_back(length-garo+2);
        answer.push_back(garo+2);
    }
    
    
    return answer;
}

File: 프로그래머스/lv2/이재표/카펫.java
import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        int total=(brown-4)/2;
        System.out.println(total);
        for(int i=1;i<=total/2;i++){
            if(i*(total-i)==yellow){
                return new int[]{total-i+2,i+2};
            }
        }
        return new int[]{0,0};
    }
}
